master/buildbot/reporters/utils.py

```diff
@@ -0,0 +1,156 @@
+        buildproperties = yield defer.gatherResults(
+            [master.data.get(("builds", build['buildid'], 'properties'))
+             for build in builds])
+    else:
+        buildproperties = range(len(builds))
+
+    if wantPreviousBuild:
+        prev_builds = yield defer.gatherResults(
+            [getPreviousBuild(master, build) for build in builds])
+    else:
+        prev_builds = range(len(builds))
+
+    if wantSteps:
+        buildsteps = yield defer.gatherResults(
+            [master.data.get(("builds", build['buildid'], 'steps'))
+             for build in builds])
+        if wantLogs:
+            for s in flatten(buildsteps, types=(list, UserList)):
+                s['logs'] = yield master.data.get(("steps", s['stepid'], 'logs'))
+                for l in s['logs']:
+                    l['content'] = yield master.data.get(("logs", l['logid'], 'contents'))
+
+    else:
+        buildsteps = range(len(builds))
+
+    for build, properties, steps, prev in zip(builds, buildproperties, buildsteps, prev_builds):
+        build['builder'] = buildersbyid[build['builderid']]
+        build['buildset'] = buildset
+        if wantProperties:
+            build['properties'] = properties
+
+        if wantSteps:
+            build['steps'] = steps
+
+        if wantPreviousBuild:
+            build['prev_build'] = prev
+
+
+# perhaps we need data api for users with sourcestamps/:id/users
+@defer.inlineCallbacks
+def getResponsibleUsersForSourceStamp(master, sourcestampid):
+    changesd = master.data.get(("sourcestamps", sourcestampid, "changes"))
+    sourcestampd = master.data.get(("sourcestamps", sourcestampid))
+    changes, sourcestamp = yield defer.gatherResults([changesd, sourcestampd])
+    blamelist = set()
+    # normally, we get only one, but just assume there might be several
+    for c in changes:
+        blamelist.add(c['author'])
+    if 'patch' in sourcestamp and sourcestamp['patch'] is not None:  # Add patch author to blamelist
+        blamelist.add(sourcestamp['patch']['author'])
+    blamelist = list(blamelist)
+    blamelist.sort()
+    defer.returnValue(blamelist)
+
+
+# perhaps we need data api for users with builds/:id/users
+@defer.inlineCallbacks
+def getResponsibleUsersForBuild(master, buildid):
+    dl = [
+        master.data.get(("builds", buildid, "changes")),
+        master.data.get(("builds", buildid, 'properties'))
+        ]
+    changes, properties = yield defer.gatherResults(dl)
+    blamelist = set()
+
+    # add users from changes
+    for c in changes:
+        blamelist.add(c['author'])
+
+    # add owner from properties
+    if 'owner' in properties:
+        blamelist.add(properties['owner'][0])
+
+    blamelist = list(blamelist)
+    blamelist.sort()
+    defer.returnValue(blamelist)
+
+
+def getURLForBuild(master, builderid, build_number):
+    prefix = master.config.buildbotURL
+    return prefix + "#builders/%d/builds/%d" % (
+        builderid,
+        build_number)
```

Comment:
No this one is entirely new code.

mail.py is arguably a rename, but has been heavily refactored